pkg/ruler/ruler.go

```diff
@@ -266,7 +266,7 @@ func buildNotifierConfig(rulerConfig *Config) (*config.Config, error) {
 	return promConfig, nil
 }
 
-func (r *Ruler) newGroup(ctx context.Context, userID string, rs []rules.Rule) (*rules.Group, error) {
+func (r *Ruler) newGroup(ctx context.Context, userID string, item *workItem) (*rules.Group, error) {
```

Comment:
I can't see the reason for this change.  You don't use `item` except to get its rules.

Reply:
This one was done for the sake of feeding something other than "default" L285. We could definitely just extract rules and filename from `item` in the caller(s) and pass them explicitly, but we'd need to add an argument to `newGroup` and modify whatever callers exist. This was the least change-y, near as we could tell.

Is it preferred that we ship both pieces of data explicitly instead?

Reply:
Sorry, I had missed `item.filename`. :+1: